backend/cfg/cfg_dominators.ml

```diff
@@ -48,15 +48,12 @@ let invariant_doms : Cfg.t -> doms -> unit =
     cfg.blocks;
   (* Check that (i) the immediate dominator is a strict dominator, and (ii)
      there is no other intermediate strict dominator - except for the entry
-     block. *)
+     block and unreachable blocks, which all have (by convention) themselves as
+     immediate dominators. *)
   Label.Tbl.iter
     (fun n idom_n ->
-      if Label.equal n cfg.entry_label
+      if not (Label.equal n idom_n)
```

Comment:
Have you considered strengthing this condition to check that `n` has no predecessors?

Reply:
I expect the condition you suggest
to be true as of today, but I wonder
whether it is guaranteed to hold.
(I am thinking about an isolated loop.)

Reply:
I am not sure if the dominator and dominator tree computation works with isolated loops, I don't think we can have them.